dmk2cw.c

```diff
@@ -372,6 +372,24 @@ approx(int a, int b)
 /* Or'ed into encoding at end of sector data */
 #define SECTOR_END 8
 
+/*
+ * Like strtol, but exit with a fatal error message if there are any
+ * invalid characters or the string is empty.
+ */
+long int
+strtol_strict(const char *nptr, int base, const char name)
+{
+  long int res;
+  char *endptr;
+
+  res = strtol(nptr, &endptr, base);
+  if (*nptr == '\0' || *endptr != '\0') {
+    fprintf(stderr, "dmk2cw: -%c requires a numeric argument\n", name);
```

Comment:
This reminds me that `dmk2cw` still needs the `fatal_msg` and related functions too.  :/

Reply:
Yeah. I ran out of time or I might have cleaned that up too.

Reply:
I looked at that some today. There is one snag in factoring out those functions and calling them from dmk2cw as well: msg has some special treatment for OUT_RAW and OUT_HEX because unlike all the other levels, OUT_RAW is not cumulative -- it omits the output that OUT_HEX adds. This special treatment is only appropriate for cw2dmk, because dmk2cw and the other functions have different sets of verbosity levels.

It would work anyway because none of the others go up as far as 7 -- but, ugh.

So I'm not going to do this until I think of a nice way to abstract it out.